certbot/main.py

```diff
@@ -875,7 +875,7 @@ def enhance(config, plugins):
     :rtype: None
 
     """
-    supported_enhancements = ["hsts", "redirect", "uir", "staple"]
+    supported_enhancements = ["auto_hsts", "hsts", "redirect", "uir"]
```

Comment:
I was planning to do this later in a enhancements - redesign PR, but it looks like it wouldn't make much sense to use the old approach here if we're planning to refactor it soon anyway. 

Using new interfaces here sounds reasonable and like a good approach which we discussed earlier. I'm going to write up a few sketches as a base for the initial discussion, which I think we should use another forum for, as keeping track of discussions in PR review comments in GitHub is rather painful :)

In short, I think the updates in renewal time should keep getting triggered from the `interfaces.GenericUpdater.generic_updates()` even if calling the enhancement interface methods to keep the changes to Certbot core code minimal when adding new enhancement types as the call chain used for renewals is complicated enough as it is.

Reply:
> I was planning to do this later in a enhancements - redesign PR, but it looks like it wouldn't make much sense to use the old approach here if we're planning to refactor it soon anyway.

Would you prefer we don't do it soon?

> In short, I think the updates in renewal time should keep getting triggered from the interfaces.GenericUpdater.generic_updates() even if calling the enhancement interface methods to keep the changes to Certbot core code minimal when adding new enhancement types as the call chain used for renewals is complicated enough as it is.

While I acknowledge, appreciate, and agree with you that tracking discussion in GH PRs is painful, I can't stop myself from providing feedback here. I hope you don't mind too much :smile:

I actually prefer we have a separate updater method for each enhancement. First, I think this allows for better documentation because the arguments each method takes can be cleanly and traditionally documented in a function rather than trying to do something like we do with `IInstaller.enhance` where we pass a dict and try and document in another file the values in the dict for each enhancement. If you were thinking of only calling `generic_updates` once for all updates, I think having separate methods helps prevent `generic_updates` from taking too many arguments over time.

Second, by having a single function, I think it causes each plugin to implement the code calling and/or determining which updates to call. Look at Apache/Nginx's implementation of `enhance`. By putting this code in Certbot, plugins don't have to implement it, simplifying their implementation and resulting in less code.

I'm certainly open to not doing things this way, but I do have a pretty strong preference for an update method per enhancement.